File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/aggregate/Enrollment.java
package cloud.ambar.creditcard.enrollment.aggregate;

import cloud.ambar.common.aggregate.Aggregate;
import lombok.Getter;
import lombok.NonNull;
import lombok.experimental.SuperBuilder;

import java.time.Instant;

@SuperBuilder(toBuilder = true)
@Getter
public class Enrollment extends Aggregate {
    @NonNull private String userId;
    @NonNull private String productId;
    @NonNull private String status;
    @NonNull private Integer annualIncomeInCents;
    @NonNull private Instant enrollmentFirstRequestedOn;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Aggregate/Enrollment.cs
namespace CreditCardEnrollment.CreditCard.Enrollment.Aggregate;

public class Enrollment : CreditCardEnrollment.Common.Aggregate.Aggregate
{
    public required string UserId { get; init; }
    public required string ProductId { get; init; }
    public required EnrollmentStatus Status { get; init; } 
    public required int AnnualIncomeInCents { get; init; }
    public required DateTime EnrollmentFirstRequestedOn { get; init; }
}